src/App.tsx

```diff
@@ -11,32 +11,64 @@ import { Footer } from "./components/Footer";
 
 import "./App.css";
 import {Contact} from "./pages/Contact";
+import {HeaderMobile} from "./components/HeaderMobile";
 
 const App: React.FC = () => {
-  return (
-    <ThemeProvider theme={theme}>
-      <Box bgcolor="primary.main" width="100%" minHeight='100vh'>
-        <HashRouter>
-          <Header />
-          <Switch >
-            <Route path={routes.home.path} exact>
-              <Home onClick={() => console.log("hey")} />
-            </Route>
-            <Route path={routes.about.path}>
-              <About />
-            </Route>
-            <Route path={routes.careers.path}>
-              <Careers />
-            </Route>
-            <Route path={routes.contact.path}>
-              <Contact />
-            </Route>
-          </Switch>
-          <Footer />
-        </HashRouter>
-      </Box>
-    </ThemeProvider>
-  );
+
+  const theme: Theme = useTheme();
+  const desktop = useMediaQuery(theme.breakpoints.up('lg'));
+
+  if (desktop) {
+    return (
+      <ThemeProvider theme={theme}>
+        <Box bgcolor="primary.main" width="100%" minHeight='100vh'>
+          <HashRouter>
+            <Header/>
+            <Switch>
+              <Route path={routes.home.path} exact>
+                <Home onClick={() => console.log("hey")}/>
+              </Route>
+              <Route path={routes.about.path}>
+                <About/>
+              </Route>
+              <Route path={routes.careers.path}>
+                <Careers/>
+              </Route>
+              <Route path={routes.contact.path}>
+                <Contact/>
+              </Route>
+            </Switch>
+            <Footer/>
+          </HashRouter>
+        </Box>
+      </ThemeProvider>
+    );
+  } else {
+    return (
+      <ThemeProvider theme={theme}>
+        <Box bgcolor="primary.main" width="100%" minHeight='100vh'>
+          <HashRouter>
+            <HeaderMobile/>
```

Comment:
Why don't we just assign HeaderMobile to a const variable and conditionally set that?

Reply:
I will try that. I tried doing something like 
`{desktop && <Header/>} 
{!desktop && <HeaderMobile/>}`
but I got the same problem.